pilot/pkg/features/pilot.go

```diff
@@ -313,7 +313,10 @@ var (
 			"It is safe to disable it if you are quite sure you don't need this feature").Get()
 
 	InjectionWebhookConfigName = env.RegisterStringVar("INJECTION_WEBHOOK_CONFIG_NAME", "istio-sidecar-injector",
-		"Name of the mutatingwebhookconfiguration to patch, if istioctl is not used.")
+		"Name of the mutatingwebhookconfiguration to patch, if istioctl is not used.").Get()
+
+	ValidationWebhookConfigName = env.RegisterStringVar("VALIDATION_WEBHOOK_CONFIG_NAME", "istio-istio-system",
```

Comment:
`istio-istio-system` is different from before, will it leak when upgrade?

Reply:
How would this be a problem on upgrade? The new control plane should look for the new resource name.

Also, this is a bit misleading but we don't actually use this value unless it's `""` (i.e. we just need this here to disable patching), the controller generates the webhook name to patch based on the revision name and Istio namespace. So the default value here is not used.

Reply:
IC, indeed very confusing

Reply:
So if this is only for patching webhookconfiguration, why not rename it to `ENABLE_PATCH_VALIDATION_WEBHOOK_CONFIG`

If the above `InjectionWebhookConfigName` is also to control patching or not, we should consider merging to one.

Reply:
Agreed... my concern was backwards compatibility since our existing external control plane guide uses these envvars. I'll ask in Environments WG if these vars have any other purpose and if not merge them into single boolean flag

Reply:
Changing it to boolean would be OK for external control plane. I can update the doc as soon as the change is merged.

Reply:
From Environments WG meeting today we should keep the existing options as they are and introduce a more formal API to control this in the future.

Reply:
And I have the TODO to figure out a way to not expose this setting in the docs (set it as the result of some other API config), but it would be good to get this PR merged ASAP as a first step, because external control plane instructions are currently broken.

Reply:
What about using `values.global.pilotCertProvider=none` to disable cert injection for both validiating and Injection webhooks. @Monkeyanator @howardjohn WDYT?